src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -389,7 +390,13 @@ private void OnBatchScopeDisposed()
 
                 foreach (var (documentId, textLoader) in documentsToClose)
                 {
-                    _workspace.ApplyChangeToWorkspace(w => w.OnDocumentClosed(documentId, textLoader));
+                    _workspace.ApplyChangeToWorkspace(w =>
+                    {
```

Comment:
I am not sure how VisualStudioProject make sure workspace doesn't mutate underneath since open/close state is mutable state on workspace.